src/views/virtualmachines/actions/components/DeleteVMModal/components/GracePeriodInput.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FC } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+  Checkbox,
+  Flex,
+  FlexItem,
+  InputGroup,
+  InputGroupText,
+  Popover,
+  StackItem,
+  TextInput,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+
+import './grace-period-input.scss';
+
+const DEFAULT_GRACE_PERIOD = 180;
+
+interface GracePeriodInputProps {
+  gracePeriodSeconds: number | null;
+  setGracePeriodSeconds: (number) => void;
+}
+
+export const GracePeriodInput: FC<GracePeriodInputProps> = ({
+  gracePeriodSeconds,
+  setGracePeriodSeconds,
+}) => {
+  const { t } = useTranslation();
+
+  const onGracePeriodCheckChange = (checked: boolean) => {
```

Comment:
I think resetting the seconds on check click is not a good practice, since in case the user clicked the checkbox by accident it will delete his previous input which can be annoying.
I would have used with another <boolean>useState for the checkbox, unless it's a requirement to reset on every check/uncheck